eras/mary/impl/cddl/lib/Cardano/Ledger/Mary/HuddleSpec.hs

```diff
@@ -0,0 +1,231 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Ledger.Mary.HuddleSpec (
+  module Cardano.Ledger.Allegra.HuddleSpec,
+  maryCDDL,
+  multiasset,
+) where
+
+import Cardano.Ledger.Allegra.HuddleSpec
+import Cardano.Ledger.Mary (MaryEra)
+import Codec.CBOR.Cuddle.Huddle
+import Data.Proxy (Proxy (..))
+import Data.Word (Word64)
+import Prelude hiding ((/))
+
+maryCDDL :: Huddle
+maryCDDL =
+  collectFrom
+    [ HIRule $ huddleRule @"block" (Proxy @MaryEra)
+    , HIRule $ huddleRule @"transaction" (Proxy @MaryEra)
+    , HIRule $ huddleRule @"policy_id" (Proxy @MaryEra)
+    , HIRule $ huddleRule @"asset_name" (Proxy @MaryEra)
+    ]
+
+instance HuddleRule "block" MaryEra where
+  huddleRule = blockRule @MaryEra
+
+instance HuddleRule "transaction" MaryEra where
+  huddleRule = transactionRule @MaryEra
+
+instance HuddleRule "header" MaryEra where
+  huddleRule = headerRule @MaryEra
+
+instance HuddleRule "header_body" MaryEra where
+  huddleRule = headerBodyRule @MaryEra
+
+instance HuddleGroup "protocol_version" MaryEra where
+  huddleGroup = protocolVersionGroup @MaryEra
+
+instance HuddleRule "major_protocol_version" MaryEra where
+  huddleRule _ = "major_protocol_version" =:= (0 :: Integer) ... (5 :: Integer)
```

Comment:
I think it would be safe to create an instance for all eras like this in cardano-ledger-core, but please confirm that it is indeed the case. If it is not, then we'll need to figure out later how to make it be the case, since that should always be true for all eras, otherwise it would be impossible to transition out of the era.
```suggestion
instance Era era => HuddleRule "major_protocol_version" era where
  huddleRule _ = "major_protocol_version" =:= (0 :: Integer) ... lastSupportedEraVersion
    where 
      lastSupportedEraVersion =
        getVersion $ errorFail (succVersion (eraProtVerHigh @era)) :: Integer)
```